rocAL/include/parameters/parameter_simple.h

```diff
@@ -35,11 +35,37 @@ class SimpleParameter : public Parameter<T> {
     T get() override {
         return _val;
     }
-    int update(T new_val) {
+
+    std::vector<T> get_array() override {
+        return _array;
+    }
+
+    void update_single_value(T new_val) {
         _val = new_val;
+    }
+
+    void update_array(T new_val) {
+        for (uint i = 0; i < _batch_size; i++) {
+            update_single_value(new_val);
```

Comment:
This seems redundant if our goal is to update one value for the complete array. 
Call update_single_value() function once and use std::fill. 

I would like to know the usecase of this function. Let's have a call to discuss this.